cmd/soroban-cli/src/rpc/transaction.rs

```diff
@@ -84,6 +92,189 @@ pub fn assemble(
     Ok(tx)
 }
 
+pub fn update_fee(
+    raw: &Transaction,
+    fee_configuration: &FeeConfiguration,
+) -> Result<Transaction, Error> {
+    let TransactionExt::V1(SorobanTransactionData { ext, resources, ..}) = &raw.ext else {
+        return Ok(raw.clone());
+    };
+    let envelope = TransactionV1Envelope {
+        tx: raw.clone(),
+        signatures: vec![
+            DecoratedSignature {
+                hint: SignatureHint([0; 4]),
+                signature: Signature::default(),
+            };
+            20
+        ]
+        .try_into()?,
+    };
+    // TODO: Deal with potentual conversion errors here
+    let tx_resources = TransactionResources {
+        instructions: resources.instructions,
+        read_entries: resources.footprint.read_only.len().try_into().unwrap(),
+        write_entries: resources.footprint.read_write.len().try_into().unwrap(),
+        read_bytes: resources.read_bytes,
+        write_bytes: resources.write_bytes,
+        metadata_size_bytes: resources.extended_meta_data_size_bytes,
+        transaction_size_bytes: envelope.to_xdr()?.len().try_into().unwrap(),
+    };
+
+    let (fee, new_refundable_fee) =
+        compute_transaction_resource_fee(&tx_resources, fee_configuration);
+    let mut tx = raw.clone();
+    // TODO: Deal with this error
+    tx.fee = tx.fee.max((fee * 115 / 100).try_into().unwrap());
+    tx.ext = TransactionExt::V1(SorobanTransactionData {
+        ext: ext.clone(),
+        resources: resources.clone(),
+        refundable_fee: new_refundable_fee,
+    });
+    Ok(tx)
+}
+
+// Use the given source_key and signers, to sign all SorobanAuthorizationEntry's in the given
+// transaction. If unable to sign, return an error.
+pub fn sign_soroban_authorizations(
+    raw: &Transaction,
+    source_key: &ed25519_dalek::Keypair,
+    signers: &[ed25519_dalek::Keypair],
+    signature_expiration_ledger: u32,
+    network_passphrase: &str,
+) -> Result<(Transaction, Vec<SorobanAuthorizationEntry>), Error> {
+    let mut tx = raw.clone();
+
+    if tx.operations.len() != 1 {
+        // This must not be an invokeHostFunction operation, so nothing to do
+        return Ok((tx, Vec::new()));
+    }
+
+    let mut op = tx.operations[0].clone();
+    let OperationBody::InvokeHostFunction(ref mut body) = &mut op.body else {
```

Comment:
FYI `CreateContract` calls now can also be signed via soroban auth, so this check is no longer correct..

Reply:
Is this still true, because CreateContract is a type of InvokeHostFunction now?

Reply:
It has always been a type of invoke host fn, but now it uses the auth payload instead of bespoke authorization mechanism.